File: src/atcoder/abc317/clojure/abc317_a.clj
(def n (read))
(def h (read))
(def x (read))
(defrecord Potion[id hp])

(def potions (map #(->Potion (inc %) (read)) (range 0 n)))
(def ans (filter #(<= x (+ h (:hp %))) potions))
(println (:id (first ans)))

File: src/atcoder/abc317/ruby/abc317_a.rb
Potion = Struct.new(:id, :hp)
n, h, x = gets.split.map(&:to_i)
potions = gets.split.map(&:to_i).zip(1..).map do |v, i|
  Potion.new i, v
end
ans = potions.select do |potion|
  x <= h + potion.hp
end
pp ans.first.id
